src/supertux/game_object_manager.cpp

```diff
@@ -196,8 +211,10 @@ GameObjectManager::flush_game_objects()
                        }
                      }),
       m_gameobjects.end());
+      s_needs_flushing_gameobjects_removed = false;
   }
 
+  if(s_needs_flushing_gameobjects_added)
   { // add newly created objects
     // Objects might add new objects in finish_construction(), so we
     // loop until no new objects show up.
```

Comment:
Don't believe this helps with optimization. `while (!m_gameobjects_new.empty())`, which is on the next line, already blocks this loop, if no new objects have been queued to be added.